test/styleq-transformProperty.test.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (c) Nicolas Gallagher
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+'use strict';
+
+import { styleq } from '../src/styleq';
+
+const propMap = {
+  marginInlineStart: ['marginInlineStart', 'marginLeft', 'marginRight'],
+  marginInlineEnd: ['marginInlineEnd', 'marginRight', 'marginLeft'],
+  paddingInlineStart: ['paddingInlineStart', 'paddingLeft', 'paddingRight'],
+  paddingInlineEnd: ['paddingInlineEnd', 'paddingRight', 'paddingLeft'],
+  marginLeft: ['marginLeft', 'marginInlineStart', 'marginInlineEnd'],
+  marginRight: ['marginRight', 'marginInlineEnd', 'marginInlineStart'],
+  paddingLeft: ['paddingLeft', 'paddingInlineStart', 'paddingInlineEnd'],
+  paddingRight: ['paddingRight', 'paddingInlineEnd', 'paddingInlineStart']
+};
+
+const transformProperty = (property) => propMap[property] || property;
+
+const styleqWithPropExpansion = styleq.factory({ transformProperty });
+
+describe('transformProperty', () => {
+  test('later overrides multiple earlier', () => {
+    const [className, style] = styleqWithPropExpansion([
+      {
+        $$css: true,
+        marginInlineStart: 'marginInlineStart-1',
+        marginInlineEnd: 'marginInlineEnd-1'
+      },
+      {
+        $$css: true,
+        marginLeft: 'marginLeft-2'
+      }
+    ]);
+    expect(className).toEqual('marginLeft-2');
+    expect(style).toEqual(null);
+  });
+
+  test('sequence of overrides', () => {
+    const [className, style] = styleqWithPropExpansion([
+      {
+        $$css: true,
+        marginInlineStart: 'marginInlineStart-1',
+        marginInlineEnd: 'marginInlineEnd-1'
+      },
+      {
+        $$css: true,
+        marginLeft: 'marginLeft-2'
+      },
+      {
+        $$css: true,
+        marginInlineStart: 'marginInlineStart-3'
+      }
+    ]);
+    expect(className).toEqual('marginInlineEnd-1 marginInlineStart-3');
+    expect(style).toEqual(null);
+  });
```

Comment:
I added this test because I'm assuming the propMap was set up to match what StyleX currently does internally? But the result is unintuitive - I think this is more signal that React Native's approach to merging directional styles is better, as you don't end up with new declarations popping into existence just because logical and physical styles are mixed in sequence.